core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultEndPoint.java

```diff
@@ -81,7 +82,16 @@ private static String buildMetricPrefix(InetSocketAddress address) {
       throw new IllegalArgumentException(
           "Could not extract a host string from provided address " + address);
     }
-    // Append the port since Cassandra 4 supports nodes with different ports
-    return hostString.replace('.', '_') + ':' + address.getPort();
+    if (address.getAddress() instanceof Inet6Address) {
+      // append getCanonicalHostName() to distinguish ip6 address from ip4
+      return hostString.replace('.', '_')
+          + ':'
+          + address.getAddress().getHostAddress()
```

Comment:
`getAddress()` may return null if the address is unresolved.

I was rather thinking about something like below:

```
    String hostString =
        address.isUnresolved() ? address.getHostString() : address.getAddress().getHostAddress();
```

IOW: if the address is unresolved, there is not much we can do about it, so use `getHostString()`; if the address is resolved however, instead of relying on the original host string as provided by the user, switch to the raw IP address, since it is unique.

I also think that using raw IP addresses whenever possible for metric prefixes may be beneficial since host names tend to create extremely long metric names (I tried recently with Astra, the metric names were a nightmare to read).

Reply:
Also, it's worth noting that apart from contact points, any other nodes (i.e., those discovered in system.peers) will have raw IP addresses anyways. So my change above would also standardize the metrics representation of contact points with that of other nodes.

Reply:
@olim7t what's your opinion on this? Is it still time to include this in 4.8?

Reply:
@olim7t what's your opinion on using raw IP addresses for metric names, including for contact points? As I see it, this would have the double benefit of fixing the original issue _and_ producing homogenous metric names.
If possible let's include this in 4.9.